parsl/providers/provider_base.py

```diff
@@ -1,7 +1,46 @@
 from abc import ABCMeta, abstractmethod, abstractproperty
+from enum import Enum
 from typing import Any, List, Optional
 
 
+class JobState(bytes, Enum):
```

Comment:
I think switching from string based job states to JobState is a useful thing to merge now, separate from the rest of this PR.